internal/attestation/azure/snp/validator.go

```diff
@@ -36,249 +41,359 @@ type Validator struct {
+			return
+		}
+
+		// https://www.amd.com/content/dam/amd/en/documents/epyc-technical-docs/specifications/57230.pdf
+		// Table 6 and 7
+		switch cert.Subject.CommonName {
+		case "SEV-Milan":
+			ask = cert
+		case "ARK-Milan":
+			ark = cert
 		default:
-			return &idKeyError{report.IDKeyDigest[:], v.config.FirmwareSignerConfig.AcceptedKeyDigests}
+			retErr = fmt.Errorf("parse certificate %d: unexpected subject CN %s", i, cert.Subject.CommonName)
+			return
 		}
+
+		i++
 	}
 
-	return nil
+	switch {
+	case i == 1:
+		retErr = fmt.Errorf("no PEM blocks found")
+		return
+	case len(rest) != 0:
+		retErr = fmt.Errorf("remaining PEM block is not a valid certificate: %s", rest)
+		return
+	}
+
+	return
 }
 
-// validateVCEKExtensions checks that the certificate extension values in cert match the values described in report.
-func validateVCEKExtensions(cert *x509.Certificate, report snpAttestationReport) error {
-	var certVersion int
-	for _, extension := range cert.Extensions {
-		switch extension.Id.String() {
-		// check bootloader version
-		case "1.3.6.1.4.1.3704.1.3.1":
-			{
-				_, err := asn1.Unmarshal(extension.Value, &certVersion)
-				if err != nil {
-					return fmt.Errorf("unmarshalling bootloader version: %w", err)
-				}
-				if certVersion != int(report.CommittedTCB.Bootloader) {
-					return fmt.Errorf("bootloader version %d from report does not match VCEK version %d", int(report.CommittedTCB.Bootloader), certVersion)
-				}
-			}
-		// check TEE version
-		case "1.3.6.1.4.1.3704.1.3.2":
-			{
-				_, err := asn1.Unmarshal(extension.Value, &certVersion)
-				if err != nil {
-					return fmt.Errorf("unmarshalling tee version: %w", err)
-				}
-				if certVersion != int(report.CommittedTCB.TEE) {
-					return fmt.Errorf("bootloader version %d from report does not match VCEK version %d", int(report.CommittedTCB.TEE), certVersion)
-				}
-			}
-		// check SNP Firmware version
-		case "1.3.6.1.4.1.3704.1.3.3":
-			{
-				_, err := asn1.Unmarshal(extension.Value, &certVersion)
-				if err != nil {
-					return fmt.Errorf("unmarshalling snp version: %w", err)
-				}
-				if certVersion != int(report.CommittedTCB.SNP) {
-					return fmt.Errorf("bootloader version %d from report does not match VCEK version %d", int(report.CommittedTCB.SNP), certVersion)
-				}
-			}
-		// check microcode version
-		case "1.3.6.1.4.1.3704.1.3.8":
-			{
-				_, err := asn1.Unmarshal(extension.Value, &certVersion)
-				if err != nil {
-					return fmt.Errorf("unmarshalling microcode version: %w", err)
-				}
-				if certVersion != int(report.CommittedTCB.Microcode) {
-					return fmt.Errorf("bootloader version %d from report does not match VCEK version %d", int(report.CommittedTCB.Microcode), certVersion)
-				}
-			}
-		}
+// parseVCEK parses the VCEK certificate from the instanceInfo into an x509-formatted certificate.
+func (a *azureInstanceInfo) parseVCEK() (vcek *x509.Certificate, retErr error) {
+	newlinesTrimmed := strings.TrimSpace(string(a.VCEK))
```

Comment:
```suggestion
	newlinesTrimmed := strings.TrimSpace(string(a.VCEK))
	if newlinesTrimmed == "" {
		return nil, nil
	}
```
Judging from the caller's usage of this func, do you want this? So that the caller doesn't log an error for an empty vcek?